airflow/ui/src/components/TriggerDag/TriggerDAGForm.tsx

```diff
@@ -224,7 +216,36 @@ const TriggerDAGForm = ({ dagId, onClose, open }: TriggerDAGFormProps) => {
                 render={({ field }) => (
                   <Field.Root mt={6}>
                     <Field.Label fontSize="md">Dag Run Notes</Field.Label>
-                    <Input {...field} placeholder="Optional" size="sm" />
+                    <Editable.Root
+                      onChange={(event: ChangeEvent<HTMLInputElement>) => {
+                        field.value = event.target.value;
+                      }}
+                      value={field.value}
+                    >
+                      <Editable.Preview
+                        _hover={{ backgroundColor: "transparent" }}
+                        alignItems="flex-start"
+                        as={VStack}
+                        gap="0"
+                        height="150px"
+                        overflowY="auto"
+                        width="100%"
+                      >
+                        {Boolean(field.value) ? (
+                          <ReactMarkdown>{field.value}</ReactMarkdown>
+                        ) : (
+                          <Text color="fg.subtle">Add a note...</Text>
+                        )}
+                      </Editable.Preview>
+                      <Editable.Textarea
+                        {...field}
+                        data-testid="notes-input"
+                        height="150px"
+                        overflowY="auto"
+                        placeholder="Add a note..."
+                        resize="none"
+                      />
+                    </Editable.Root>
```

Comment:
Good idea.

Ideally I think we want to extract a component from this `Editable/Preview' for markdown content input.

I feel like this will be needed to a different places.